tools/cjxl_ng_main.cc

```diff
@@ -695,49 +795,83 @@ int main(int argc, char** argv) {
       std::cerr << "No frames on input file.\n";
       return EXIT_FAILURE;
     }
-    const jxl::extras::PackedFrame& pframe = ppf.frames[0];
-    const jxl::extras::PackedImage& pimage = pframe.color;
-    JxlPixelFormat ppixelformat = pimage.format;
 
     {  // JxlEncoderSetBasicInfo
       JxlBasicInfo basic_info;
+      const JxlBasicInfo& ppf_info = ppf.info;
       JxlEncoderInitBasicInfo(&basic_info);
-      basic_info.xsize = pimage.xsize;
-      basic_info.ysize = pimage.ysize;
-      basic_info.bits_per_sample = 32;
-      basic_info.exponent_bits_per_sample = 8;
-      basic_info.uses_original_profile = JXL_FALSE;
+      basic_info.xsize = ppf_info.xsize;
+      basic_info.ysize = ppf_info.ysize;
+      basic_info.bits_per_sample = ppf_info.bits_per_sample;
+      basic_info.exponent_bits_per_sample = ppf_info.exponent_bits_per_sample;
+      basic_info.have_animation = ppf_info.have_animation;
+      basic_info.animation.tps_numerator = ppf_info.animation.tps_numerator;
+      basic_info.animation.tps_denominator = ppf_info.animation.tps_denominator;
+      basic_info.animation.num_loops = ppf_info.animation.num_loops;
+
+      basic_info.num_extra_channels =
+          (ppf.frames[0].color.format.num_channels == 2 ||
+           ppf.frames[0].color.format.num_channels == 4)
+              ? 1
+              : 0;
+      basic_info.uses_original_profile = JXL_TRUE;
```

Comment:
doing it that way does not work on animations without alpha, I think. Did you test that?
I suggest to use the `ppf.info` directly without making another `basic_info`, basically like here:
https://github.com/mo271/libjxl/commit/18703a0312b6af8e4832d13a68ae2abb4f751eae

Hardcoding 'uses_original_profile` will lead to file size increase.

Reply:
Also currently this does not work with animation containing frame with frame dimensions less than the dimension of the image. I think we should either catch that (rather common) case and give a meaningful error or fix it

Reply:
This needs tests anyhow. Overall, we are still in "let's get this feature-complete" phase. There will have to be a cleanup afterwards.